addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -76,7 +76,7 @@ const TableOfContent = publicWidget.Widget.extend({
      * @override
      */
     destroy() {
-        this._scrollTarget.removeEventListener("scroll", this._onScrollBound);
+        this._scrollTarget?.removeEventListener("scroll", this._onScrollBound);
```

Comment:
Your fix works, but I am not sure it is the right one :open_mouth: 
Because here you simply ignore the issue, but the listeners added by the start are well there, so we have four listeners when in edit mode, instead of two:
the two 2 added by the `start` > go in edit mode > `destroy` does not remove them > `start` adds 2 more

I think we should rather find why it is undefined when this specific destroy is called, it seems that something is not properly awaited so my guess is that it is a race condition or something like that :slightly_smiling_face: 
I did not investigate deeply, but by printing the sequence, we have something like that :arrow_down: 
![image](https://github.com/user-attachments/assets/360c24b0-a658-4fc3-8141-8f36154c2dba)

As you can see, it seems that the `start` is called (in red), then `destroy` is called (in green) and then the rest of the `start` executes (red), so this is why the `scrollTarget` was undefined: because the start did not finish before `destroy was called` :open_mouth: 

Maybe investigate this instead :slightly_smiling_face:

Reply:
> it seems that something is not properly awaited

Indeed, waiting until all the `start` is completed solves the issue.